src/components/PeriodSelector/modules/FixedPeriodsGenerator.js

```diff
@@ -62,7 +84,12 @@ function WeeklyPeriodType(formatYyyyMmDd, weekObj, fnFilter) {
             period.id = period.iso
             date.setDate(date.getDate() + 6)
             period.endDate = formatYyyyMmDd(date)
-            period.name = `Week ${week} - ${period.startDate} - ${period.endDate}`
+
+            const weekNumber = week
```

Comment:
`const` needed? same for biWeekNumber, monthName etc below

Reply:
Yes, it is needed. A new const (`weekNumber`) is needed for each iteration in the loop, otherwise, the name function will use the latest value of `week` when the name function is later executed. Same goes for all these other `const` vars that I've introduced. Otherwise, you end up with this:

![image](https://user-images.githubusercontent.com/6113918/81593231-ff2eca00-9373-11ea-9308-7f4f6f592d3a.png)

Reply:
If I don't have a `const monthName`:

![image](https://user-images.githubusercontent.com/6113918/81593643-9bf16780-9374-11ea-977f-059748e91d6a.png)

Reply:
What I mean for these constants (`weekNumber`, `biWeekNumber` `monthName`) is that they're only used in one single place, like this example
`${i18n.t('Week {{weekNumber}}', { weekNumber })} - ${period.startDate} - ${period.endDate}` (https://github.com/dhis2/analytics/blob/3f5060970e26044fd2bdb0e89c1d59dbdbae202d/src/components/PeriodSelector/modules/FixedPeriodsGenerator.js#L90).

Can this be replaced with just 
`${i18n.t('Week {{weekNumber}}', { weekNumber: week })} - ${period.startDate} - ${period.endDate}` 
to avoid creating a single-use `const`?

Reply:
regarding `monthName`, you do indeed have a const named `monthName`, it's here: https://github.com/dhis2/analytics/blob/3f5060970e26044fd2bdb0e89c1d59dbdbae202d/src/components/PeriodSelector/modules/FixedPeriodsGenerator.js#L188